onebusaway-android/src/main/java/org/onebusaway/android/tripservice/PollerTask.java

```diff
@@ -120,55 +121,51 @@ private void poll1(Cursor c) {
 
         final String tripId = c.getString(COL_TRIP_ID);
         final String stopId = c.getString(COL_STOP_ID);
-        final long reminderMS = getReminderMS(tripId, stopId);
+        final long reminderMin = getReminderMin(tripId, stopId);
 
         ObaArrivalInfoResponse response = ObaArrivalInfoRequest
                 .newRequest(mContext, stopId).call();
 
-        Long departMS = null;
+        // Arrival information
+        ArrivalInfo arrivalInfo = null;
         if (response.getCode() == ObaApi.OBA_OK) {
-            departMS = checkArrivals(response, c);
+            arrivalInfo = checkArrivals(response, c);
         }
 
-        if (departMS != null) {
-            final long diffTime = departMS - System.currentTimeMillis();
-            if (diffTime <= reminderMS) {
+        if (arrivalInfo != null) {
+            if (arrivalInfo.getEta() <= reminderMin) {
                 // Bus is within the reminder interval (or it possibly has left!)
                 // Send off a notification.
                 //Log.d(TAG, "Notify for trip: " + alertUri);
-                TripService.notifyTrip(mContext, mUri, diffTime);
+                TripService.notifyTrip(mContext, mUri, arrivalInfo.getNotifyText());
             }
         }
     }
 
-    private long getReminderMS(String tripId, String stopId) {
+    private long getReminderMin(String tripId, String stopId) {
         final Uri uri = ObaContract.Trips.buildUri(tripId, stopId);
-        return (long) UIUtils.intForQuery(mContext, uri, ObaContract.Trips.REMINDER) * ONE_MINUTE;
+        return (long) UIUtils.intForQuery(mContext, uri, ObaContract.Trips.REMINDER);
     }
 
-    //
-    // Return the difference between now and the predicted/scheduled
-    // arrival time, or null if the arrival can't be found.
-    //
-    private Long checkArrivals(ObaArrivalInfoResponse response, Cursor c) {
+    /**
+     * Checks arrivals from given ObaArrivalInfoResponse
+     *
+     * @param response arrival information
+     * @param c        holds the stored trip_id
+     * @return ArrivalInfo, or return null if the arrival can't be found.
+     */
+    private ArrivalInfo checkArrivals(ObaArrivalInfoResponse response, Cursor c) {
```

Comment:
I think we should pass in the `tripId` here, rather than the `Cursor c`.  It provides better separation from the underlying database.